zilliqa/src/consensus.rs

```diff
@@ -37,9 +37,8 @@ use crate::{
 
 #[derive(Debug)]
 struct NewViewVote {
-    signatures: Vec<NodeSignature>,
-    signers: Vec<u16>,
-    cosigned: BitVec,
+    // Keeps signer index in deposit contract and sent signature
+    signatures: Vec<(u16, NodeSignature)>,
```

Comment:
I would put this in a structure for readability. Otherwise, particularly with type inference, everyone is going to lose track of which order the tuple is in fairly quickly and then we'll get confused, and ..

Also, why store this as a vector? Given the ops you're doing on it later on, would a HashMap<> or a BTreeMap<> be faster?